### PR TITLE
GIVCAMP-88 | data card

### PR DESCRIPTION
**Pull request by yvonnetangsu**
# READY FOR REVIEW

# Summary
- Add data card with animated counter option

# Review By (Date)
- Retro

# Criticality
- 4

# Review Tasks

## Setup tasks and/or behavior to test

1. Go to storyblok test/test-initiative-template page
2. Select the PR245 preview URL in the visual editor
3. Scroll down to see data cards in dark and light mode.
Screenshot 2024-03-12 at 1 45 22 PM
Screenshot 2024-03-12 at 1 45 28 PM

4. Check that the cards with number counter enabled, the number in the heading counts up once the card comes into view.

# Associated Issues and/or People
- JIRA ticket(s) - GIVCAMP-88

### CODE REVIEW

File: components/DataCard/DataCard.tsx [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+        width="full"
+        pt={paddingTop}
+        className={styles.root}
+        {...props}
+      >
+        <FlexBox direction="col" className={styles.flex}>
+          {/* If number counter is enabled, aria-hidden the animated heading and add a SR only heading */}
+          {isCounter && heading && (
+            <Heading as={headingLevel} srOnly>{heading}</Heading>
+          )}
```
Comment (yvonnetangsu) [inline components/DataCard/DataCard.tsx:57]: If there is a number counter in the heading, add a SR only heading with all static heading, then aria-hidden the visible heading with the number counter.

File: README.md [hunk 2]
```diff
@@ -1,11 +1,11 @@
-# [Giving Campaign (Stanford On Purpose)](https://github.com/SU-SWS/ood-giving-campaign)
+# [Stanford Momentum](https://github.com/SU-SWS/ood-giving-campaign)
```
Comment (yvonnetangsu) [inline README.md:1]: Updating some readme

File: tailwind/plugins/theme/gc-lineHeight.js [hunk 3]
```diff
@@ -0,0 +1,9 @@
+/**
+ * Momentum line heights
+ */
+module.exports = function () {
+  return {
+    // Extra tight line height for Druk font in some components, e.g., Data Card, Moment Poster
+    druk: '0.9',
+  };
+};
```
Comment (yvonnetangsu) [inline tailwind/plugins/theme/gc-lineHeight.js:9]: Add extra tight leading utility for Druk font that are used in a few places

File: components/DataCard/DataCard.styles.ts [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+
+export const flex = 'h-full';
+export const content = (
+  hasBarColor?: boolean,
+) => cnb('rs-pl-2', {
+  'border-l-[1.4rem] md:border-l-[2rem]': hasBarColor,
+});
+
+export const heading = 'rs-mb-3 ml-22 whitespace-pre-line mt-auto';
+export const body = '*:*:leading-snug';
```
Comment (sherakama) [inline components/DataCard/DataCard.styles.ts:15]: 👀

Comment (yvonnetangsu) [reply components/DataCard/DataCard.styles.ts:15]: It's a WYSIWYG 🤣

**Comment by yvonnetangsu** [comment]
@sherakama I think I'm going to merge this and cut a release. It doesn't affect anything that's published so it's pretty safe 😄 If you see anything you're concerned about LMK

**Pull request merged**